src/ui/server/controllers/insurance/business/company-details/index.POST.test.ts

```diff
@@ -36,151 +35,35 @@ describe('controllers/insurance/business/companies-details', () => {
-        [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
-      };
-
-      const getCompaniesHouseResponse = jest.fn(() => Promise.resolve(mockCompany));
-      api.keystone.getCompaniesHouseInformation = getCompaniesHouseResponse;
-
-      await post(req, res);
-
-      const errorMessageTradingName = EXPORTER_BUSINESS_ERROR[TRADING_NAME].IS_EMPTY;
-
-      expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
-        ...corePageVariables({
-          PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
-          BACK_LINK: req.headers.referer,
-        }),
-        ...pageVariables(mockApplication.referenceNumber),
-        validationErrors: generateValidationErrors(TRADING_NAME, errorMessageTradingName, {}),
-        submittedValues,
-      });
-    });
-
-    it('should display validation error for companies house when company house error but trading name radio is selected', async () => {
-      req.body = {
-        companiesHouseNumber: '123456',
-        [TRADING_NAME]: 'true',
-      };
-
-      const submittedValues = {
-        [COMPANY_HOUSE.INPUT]: req.body[COMPANY_HOUSE.INPUT],
-        [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
-      };
-
-      const getCompaniesHouseResponse = jest.fn(() => Promise.resolve({ apiError: true }));
-      api.keystone.getCompaniesHouseInformation = getCompaniesHouseResponse;
-
-      await post(req, res);
-
-      const errorMessage = EXPORTER_BUSINESS_ERROR[COMPANY_HOUSE.INPUT].TECHNICAL_ISSUES;
-
-      expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
-        ...corePageVariables({
-          PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
-          BACK_LINK: req.headers.referer,
-        }),
-        ...pageVariables(mockApplication.referenceNumber),
-        validationErrors: generateValidationErrors(COMPANY_HOUSE.INPUT, errorMessage, {}),
-        submittedValues,
+    const errorMessageTradingName = EXPORTER_BUSINESS_ERROR[TRADING_NAME].IS_EMPTY;
+    const errorMessageTradingAddress = EXPORTER_BUSINESS_ERROR[TRADING_ADDRESS].IS_EMPTY;
+    describe('when there are validation errors', () => {
+      it('should render template with validation errors', async () => {
+        req.body = {};
+
+        const submittedValues = {
+          [COMPANY_HOUSE.INPUT]: req.body[COMPANY_HOUSE.INPUT],
+          [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
+          [TRADING_ADDRESS]: sanitiseValue(req.body[TRADING_ADDRESS]),
+        };
+
+        await post(req, res);
+
+        const errorMessageCompanyHouseIncorrect = EXPORTER_BUSINESS_ERROR[COMPANY_HOUSE.INPUT].INCORRECT_FORMAT;
+
+        let validationErrors = generateValidationErrors(COMPANY_HOUSE.INPUT, errorMessageCompanyHouseIncorrect, {});
+        validationErrors = generateValidationErrors(TRADING_NAME, errorMessageTradingName, validationErrors);
+        validationErrors = generateValidationErrors(TRADING_ADDRESS, errorMessageTradingAddress, validationErrors);
```

Comment:
Shouldn't a single call to `generateValidationErrors` with form data object, generate errors for all the fields?

Reply:
yes but it is looping through the errors and calling it a single time each loop? `generateValidationErrors = (fieldId: string, errorText: string, errors: ValidationErrors = { errorList: {}, summary: [] })`

Reply:
I see - in the test it's called the generic helper function fora  single field, but the controller doesn't call this:

```js
validationErrors = companyDetailsValidation(body, validationErrors);
```

Can switch to this in the test?